src/prefect/tasks/azure/blobstorage.py

```diff
@@ -0,0 +1,144 @@
+import uuid
+
+import azure.storage.blob
+
+from prefect import Task
+from prefect.client import Secret
+from prefect.utilities.tasks import defaults_from_attrs
+
+
+class BlobStorageDownload(Task):
+    """
+    Task for downloading data from an Blob Storage container and returning it as a string.
+    Note that all initialization arguments can optionally be provided or overwritten at runtime.
+
+    Args:
+        - azure_credentials_secret (str, optional): the name of the Prefect Secret
+            that stores your Azure credentials; this Secret must be a JSON string
+            with two keys: `ACCOUNT_NAME` and `ACCOUNT_KEY`
+        - container (str, optional): the name of the Azure Blob Storage to download from
+        - **kwargs (dict, optional): additional keyword arguments to pass to the
+            Task constructor
+    """
+
+    def __init__(
+        self,
+        azure_credentials_secret: str = "AZ_CREDENTIALS",
+        container: str = None,
+        **kwargs
+    ) -> None:
+        self.azure_credentials_secret = azure_credentials_secret
+        self.container = container
+        super().__init__(**kwargs)
+
+    @defaults_from_attrs("azure_credentials_secret", "container")
+    def run(
+        self,
+        blob_name: str,
+        azure_credentials_secret: str = "AZ_CREDENTIALS",
+        container: str = None,
+    ) -> str:
+        """
+        Task run method.
+
+        Args:
+            - blob_name (str): the name of the blob within this container to retrieve
+            - azure_credentials_secret (str, optional): the name of the Prefect Secret
+                that stores your Azure credentials; this Secret must be a JSON string
+                with two keys: `ACCOUNT_NAME` and `ACCOUNT_KEY`
+            - container (str, optional): the name of the Blob Storage container to download from
+
+        Returns:
+            - str: the contents of this blob_name / container, as a string
+        """
+
+        if container is None:
+            raise ValueError("A container name must be provided.")
+
+        # get Azure credentials
+        azure_credentials = Secret(azure_credentials_secret).get()
+        az_account_name = azure_credentials["ACCOUNT_NAME"]
+        az_account_key = azure_credentials["ACCOUNT_KEY"]
+
+        blob_service = azure.storage.blob.BlockBlobService(
+            account_name=az_account_name, account_key=az_account_key
+        )
+
+        blob_result = blob_service.get_blob_to_text(
+            container_name=container, blob_name=blob_name
+        )
+        content_string = blob_result.content
+
+        # TODO: some other format?
+        return content_string
```

Comment:
What does the `blob_result` look like without grabbing `.content`?

Reply:
`get_blob_to_text` returns a [`azure.storage.blob.models.Blob`](https://docs.microsoft.com/en-us/python/api/azure-storage-blob/azure.storage.blob.models.blob?view=azure-python) object.